pkg/istio-agent/xds_proxy.go

```diff
@@ -224,6 +231,25 @@ func (p *XdsProxy) StreamAggregatedResources(downstream discovery.AggregatedDisc
 				proxyLog.Errorf("upstream send error for type url %s: %v", req.TypeUrl, err)
 				return err
 			}
+		case healthEvent := <-healthEventsChan:
```

Comment:
This is going to get triggered when healthEventsChan closes. See https://play.golang.org/p/hiSK2_PgNS5. We should probably check for `ok`